cli/ts/commands/joinPoll.ts

```diff
@@ -0,0 +1,386 @@
+ * @param maciPrivKey User's private key for signing up
+ * @param stateLeafIndex Index where the user is stored in the state leaves
+ * @param credits Credits for voting
+ * @param pollPrivKey Poll's private key for the poll joining
+ * @param pollPubKey Poll's public key for the poll joining
+ * @returns stringified circuit inputs
+ */
+const joiningCircuitInputs = (
+  signUpData: IGenSignUpTree,
+  stateTreeDepth: bigint,
+  maciPrivKey: PrivKey,
+  stateLeafIndex: bigint,
+  credits: bigint,
+  pollPrivKey: PrivKey,
+  pollPubKey: PubKey,
+): IPollJoiningCircuitInputs => {
+  // Get the state leaf on the index position
+  const { signUpTree: stateTree, stateLeaves } = signUpData;
+  const stateLeaf = stateLeaves[Number(stateLeafIndex)];
+  const { pubKey, voiceCreditBalance, timestamp } = stateLeaf;
+  const pubKeyX = pubKey.asArray()[0];
+  const pubKeyY = pubKey.asArray()[1];
+  const stateLeafArray = [pubKeyX, pubKeyY, voiceCreditBalance, timestamp];
+  const pollPubKeyArray = pollPubKey.asArray();
+
+  assert(credits <= voiceCreditBalance, "Credits must be lower than signed up credits");
+
+  // calculate the path elements for the state tree given the original state tree
+  const { siblings, index } = stateTree.generateProof(Number(stateLeafIndex));
+  const siblingsLength = siblings.length;
+
+  // The index must be converted to a list of indices, 1 for each tree level.
+  // The circuit tree depth is this.stateTreeDepth, so the number of siblings must be this.stateTreeDepth,
+  // even if the tree depth is actually 3. The missing siblings can be set to 0, as they
+  // won't be used to calculate the root in the circuit.
+  const indices: bigint[] = [];
+
+  for (let i = 0; i < stateTreeDepth; i += 1) {
+    // eslint-disable-next-line no-bitwise
+    indices.push(BigInt((index >> i) & 1));
+
+    if (i >= siblingsLength) {
+      siblings[i] = BigInt(0);
+    }
+  }
+
+  const siblingsArray = siblings.map((sibling) => [sibling]);
+
+  // Create nullifier from private key
+  const inputNullifier = BigInt(maciPrivKey.asCircuitInputs());
+  const nullifier = poseidon([inputNullifier]);
+
+  // Get pll state tree's root
+  const stateRoot = stateTree.root;
+
+  // Set actualStateTreeDepth as number of initial siblings length
+  const actualStateTreeDepth = BigInt(siblingsLength);
+
+  // Calculate public input hash from nullifier, credits and current root
+  const inputHash = sha256Hash([nullifier, credits, stateRoot, pollPubKeyArray[0], pollPubKeyArray[1]]);
+
+  const circuitInputs = {
+    privKey: maciPrivKey.asCircuitInputs(),
+    pollPrivKey: pollPrivKey.asCircuitInputs(),
+    pollPubKey: pollPubKey.asCircuitInputs(),
+    stateLeaf: stateLeafArray,
+    siblings: siblingsArray,
+    indices,
+    nullifier,
+    credits,
+    stateRoot,
+    actualStateTreeDepth,
+    inputHash,
+  };
+
+  return stringifyBigInts(circuitInputs) as unknown as IPollJoiningCircuitInputs;
+};
+
+export const joinPoll = async ({
```

Comment:
Let's add docs for all the exported methods here

Reply:
https://github.com/0x3327/maci/commit/03fa22b92075afb8d2dc261a3af78282446c4197